deps/build.jl

```diff
@@ -2,53 +2,73 @@
     using Libdl
 end
 
-depsfile = joinpath(dirname(@__FILE__),"deps.jl")
+depsfile = joinpath(dirname(@__FILE__), "deps.jl")
 if isfile(depsfile)
     rm(depsfile)
 end
 
 function write_depsfile(path)
-    open(depsfile,"w") do f
-        print(f,"const libcplex = ")
+    open(depsfile, "w") do f
+        print(f, "const libcplex = ")
         show(f, path) # print with backslashes excaped on windows
         println(f)
     end
 end
 
-@static if (VERSION >= v"0.7.0-DEV.3382" && Sys.isapple()) || (VERSION < v"0.7.0-DEV.3382" && is_apple())
-    Libdl.dlopen("libstdc++",Libdl.RTLD_GLOBAL)
+@static if Sys.isapple()
+    Libdl.dlopen("libstdc++", Libdl.RTLD_GLOBAL)
 end
 
 base_env = "CPLEX_STUDIO_BINARIES"
 
-const cpxvers = ["128", "1280", "129", "1290"]
+cplex_path = try
+    # Find the path to the CPLEX executable.
+    @static if Sys.isapple() || Sys.isunix()
+        read(`which cplex`, String)
+    elseif Sys.iswindows()
+        read(`where cplex`, String)
+    end
+catch
+    nothing
+end
+if cplex_path !== nothing
+    # Extract the path to the folder containing the CPLEX executable.
+    cplex_path = dirname(strip(cplex_path))
+end
+
+base_cpxvers = ["128", "129"]
+cpxvers = [base_cpxvers; base_cpxvers .* "0"]
 
 libnames = String["cplex"]
 for v in reverse(cpxvers)
-    if (VERSION >= v"0.7.0-DEV.3382" && Sys.isapple()) || (VERSION < v"0.7.0-DEV.3382" && is_apple())
+    if Sys.isapple()
         push!(libnames, "libcplex$v.dylib")
-    elseif (VERSION >= v"0.7.0-DEV.3382" && Sys.isunix()) || (VERSION < v"0.7.0-DEV.3382" && is_unix())
+    elseif Sys.isunix()
         push!(libnames, "libcplex$v.so")
         if haskey(ENV, base_env)
             push!(libnames, joinpath(ENV[base_env], "libcplex$v.so"))
         end
+        if cplex_path !== nothing
+            push!(libnames, joinpath(cplex_path, "libcplex$v.so"))
+        end
```

Comment:
Why only add this is `isiunix`? What about the `is apple` and the `.dylib`?

Reply:
I did something in the latest commit, but I have no possibility to test on macOS… Does that look right to you? (Maybe I should merge the two `@static if`?)

Reply:
Yeah, can't this whole file just be tidied up and simplified?